### ISSUE TITLE
Heads-up: Helm will get a push subcommand

### ISSUE THREAD

**Issue by lindhe** [chartmuseum/helm-push#103]
I just wanted to give you a heads-up that [the implementation](https://github.com/helm/helm/pull/9782) of [HIP 6](https://github.com/helm/community/blob/main/hips/hip-0006.md#6-helm-chart-is-removed-integrated-into-rest-of-cli) will introduce a `push` subcommand to Helm. I guess that will conflict with your plugin, and perhaps the plugin must be renamed in order to keep working.

**Comment by tomislater** [comment]
Yeah, I encountered on this issue today.

```
$ helm plugin install https://github.com/chartmuseum/helm-push.git
Downloading and installing helm-push v0.9.0 ...
https://github.com/chartmuseum/helm-push/releases/download/v0.9.0/helm-push_0.9.0_linux_amd64.tar.gz
Installed plugin: push
failed to load plugins: two plugins claim the name "push" at "/root/.local/share/helm/plugins/helm-push" and "/root/.local/share/helm/plugins/helm-push.git"
```

**Comment by lindhe** [comment]
@tomislater which `helm version` are you on?

**Comment by tomislater** [comment]
The newest one: https://github.com/helm/helm/releases/tag/v3.6.3

**Comment by tomislater** [comment]
Nevermind... I use `alpine/k8s` image, and `helm-push` plugin has been added to this image lately: https://github.com/alpine-docker/k8s/commits/master

So, that's my problem. I just do not need to install this plugin twice.

**Comment by lindhe** [comment]
[It](https://github.com/helm/helm/pull/9782) is merged now, and expected to launch with [Helm 3.7](https://github.com/helm/helm/milestone/111) soon.

**Comment by sathieu** [comment]
Now Helm 3.7.0 was released, and `helm push` now fails with:

```
$ helm push foo.tgz repo
Error: this feature has been marked as experimental and is not enabled by default. Please set HELM_EXPERIMENTAL_OCI=1 in your environment to use this feature
```

(from https://gitlab.com/kubitus-project/external-helm-charts/-/jobs/1594830481)

**Comment by gillesde** [comment]
Created a related issue regarding the failures of `helm push`: https://github.com/chartmuseum/helm-push/issues/109

**Comment by thomastvedt** [comment]
Damn, all my CI/CD builds stopped working now when Helm 3.7 hides or overrides my old push plugin 🙈
Was it in version 3.7 that Helm got it's own push command?

I tried using the "new" helm push file.tgz target, but it doesn't support http or https, only oci://?

**Comment by lindhe** [comment]
Yes, it was introduced in Helm 3.7. More details here: https://github.com/helm/helm/releases/tag/v3.7.0

Yeah, the new push command only supports OCI since there is no push mechanism for the old variant.

**Comment by thomastvedt** [comment]
Ok, so what is "OCI", and does [chartmuseum](https://github.com/helm/chartmuseum) support it? :)   
Should I migrate to use helms new built in command?  
Or is there a workaround to continue to use `helm-push`?

**Comment by gillesde** [comment]
> Ok, so what is "OCI", and does [chartmuseum](https://github.com/helm/chartmuseum) support it? :) Should I migrate to use helms new built in command? Or is there a workaround to continue to use `helm-push`?

@thomastvedt  A simple alternative is to revert to Helm 3.6.3 for the moment, i.e. modify your D/CI pipelines to the following:
```
# Install Helm 3.6.3
curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 > get_helm.sh
chmod 700 get_helm.sh
./get_helm.sh --version v3.6.3
helm version
```

**Comment by jdolitsky** [comment]
Hello, the latest version of the plugin (v0.10.0 released 2 days ago) has renamed the subcommand to `cm-push` so as not to conflict with Helm @rex

**Comment by thomastvedt** [comment]
Yep, the plugin is renamed.. After reverting to 3.6.3 my builds started failing again, and that was the plugin with a new name of course 🙈 I guess this issue can be closed then?

**Comment by lindhe** [comment]
As far as I'm concerned, this issue was resolved by d7cc009a34f06eea26f87a03566a2c3f246d4394 so I'm marking this as closed.

Maintainers may reopen this for visibility if desired.
